### PR TITLE
[feat]: Adds double click handling to PreviewCanvas

### PR DESCRIPTION
**Pull request by allishultes**
**Is your Pull Request request related to another [issue](https://github.com/bbc/digital-paper-edit-storybook/issues) in this repository ?**      
Related to [issue #85 in DPE](https://github.com/bbc/digital-paper-edit-client/issues/85).

**Describe what the PR does**    

- [x] Adds a prop for `currentTime` that can be passed to PreviewCanvas when a user double clicks on a word
- [x] Sets video context playback time to be equal to the currentTime value when it changes

Out of scope / not completed:
- [ ] Storybook showcase of this prop in action
- [ ] There is an ongoing issue with the quality of the media in DPE PreviewCanvas which is being addressed in DPE issue 40. 

**State whether the PR is ready for review or whether it needs extra work**    

Ready for review. To test, you need to move the PreviewCanvas folder to your local DPE setup, and switch to `i85-double-click`. 

**Additional context**

### CODE REVIEW

**Comment by emettely** [comment]
The code looks small enough and looks ok to me, but to demo this feature you need to add it to the storybook in here: http://localhost:6006/?path=/story/programmescripteditor-not-published--default

In order to actually demo this, you'd need to:

1. Add dummy timing data, if not there already to the `import { programmeScript } from '../../dummy';`
2. Clicking text shows some feedback and
3. Clicking text does the feature

Steps 2 + 3 can be copied over from the firebase repo.

File: src/PreviewCanvas/VideoContextPreview/index.js [hunk 1]
```diff
@@ -51,6 +51,18 @@ const VideoContextPreview = (props) => {
 
   }, [ props.playlist, videoContext ]);
 
+  useEffect(() => {
+    if (currentTime) {
+
+      setVideoContext(vc => {
+        vc.currentTime = currentTime;
+        videoContext.play();
```
Comment (emettely) [inline src/PreviewCanvas/VideoContextPreview/index.js:59]:
```suggestion
        vc.play();
```

File: src/ProgrammeScriptContainer/index.js [hunk 2]
```diff
@@ -1,18 +1,36 @@
...
+import Card from 'react-bootstrap/Card';
 import arrayMove from 'array-move';
 import { SortableContainer, } from 'react-sortable-hoc';
 import ProgrammeElements from './ProgrammeElements';
+import PreviewCanvas from '../PreviewCanvas';
 
 const ProgrammeScriptContainer = (props) => {
   const [ items, setItems ] = useState(props.items);
+  const [ timeOnClick, setTimeOnClick ] = useState();
+  const width = useState(150);
```
Comment (emettely) [inline src/ProgrammeScriptContainer/index.js:12]: does this make the video go narrow?
```suggestion
  const width = useState(300);
```

**Comment by allishultes** [comment]
PaperEdit: 
```
{
  start: // used for VC
  end: 
  words: [
    start: // need to be made relative to VC 
    end: 
  ],
  vcStart: // start time of a clip within vc (new field)
  vcEnd: 
}
```

Word times need to be recalculated:
- [ ] When a paper-cut is added to the programmeScript (the addition plus all paper-cuts later in the script; in the last case, by the duration of the added clip)
    - To get new paper-cut timings:
      - [ ] Get the `vc.end` time of the paper-cut previous in the script. This becomes its `vc.start`. 
      - [ ] Words' new start times = (paperEdit.word.start - paperEdit.start) + vc.Start
      - [ ] Words' new end times = start time + duration of word.
    - To modify paper-cut times following:
      - [ ] Add the duration of the new paper-cut to each of the words' start and end times.
    - Index values also should be updated?? (I'm not sure what these are used for tbh)
- [ ] When the programme script order is changed (everything between the clips being re-ordered? If the clips don't change in length, the video before and after the reordered elements should be the same)
  - [ ] Slice items between the indexes of changed paper-cuts.
  - [ ] Recalculate word timings, vc.start and vc.end, plus indexes.
- [ ] When an item is deleted (all words subtract that clip's duration from the start-time)
    - [ ] All words after that paper-cut index should subtract the duration of that paper-cut from their words' start and end times, as well as their vc.start and vc.end times.

**Pull request closed without merging**